cedar-policy/src/frontend/utils.rs

```diff
@@ -98,14 +163,10 @@ impl InterfaceResult {
 
 #[cfg(test)]
 #[track_caller] // report the caller's location as the location of the panic, not the location in this function
-pub(crate) fn assert_is_failure(result: &InterfaceResult, internal: bool, err: &str) {
+pub(crate) fn assert_is_failure(result: &InterfaceResult, internal: bool) {
     use cool_asserts::assert_matches;
-    use itertools::Itertools;
 
-    assert_matches!(result, InterfaceResult::Failure { is_internal, errors } => {
-        assert!(
-            errors.iter().exactly_one().unwrap().contains(err),
```

Comment:
Can we keep this assertion?

Reply:
I unified the code used to parse policies in `is_authorized.rs` and `validate.rs`, which resulted in some changed error messages (in particular, there was no longer exactly one error in some cases). Was easier to delete the assertion than alter it 😆 

I'll work on adding it back this afternoon.

Reply:
Revised test cases. Lmk what you think.